concourse/steps/update_component_deps.py

```diff
@@ -205,18 +205,44 @@ def determine_reference_versions(
         raise NotImplementedError
 
 
+def create_ctx_repo_mapping(
+    map: dict[str, str],
+) -> dict[str, gci.componentmodel.OciRepositoryContext]:
+    # invert the input map and replace the ctx repo names with actual instances, i.e. turn it from
+    # a mapping of ctx repo name -> component name(s) to a mapping of component name -> ctx repo.
+    cfg_factory = ci.util.ctx().cfg_factory()
+
+    return {
+        component_name: gci.componentmodel.OciRepositoryContext(
+            baseUrl=cfg_factory.ctx_repository(ctx_repo_name).base_url()
+        )
+        for ctx_repo_name, component_name_list in map.items()
+            for component_name in component_name_list
+    }
+
+
 def determine_upgrade_prs(
     upstream_component_name: str,
     upstream_update_policy: UpstreamUpdatePolicy,
     upgrade_pull_requests: typing.Iterable[github.util.UpgradePullRequest],
-    ctx_repo: gci.componentmodel.OciRepositoryContext,
+    default_ctx_repo: gci.componentmodel.OciRepositoryContext,
+    ctx_repo_mapping: dict[str, gci.componentmodel.OciRepositoryContext],
     ignore_prerelease_versions=False,
-) -> typing.Iterable[typing.Tuple[
-    gci.componentmodel.ComponentReference, gci.componentmodel.ComponentReference, str
-]]:
+) -> typing.Iterable[
+    typing.Tuple[
+        gci.componentmodel.ComponentReference,
```

Comment:
btw, how about extending `ComponentReference` to also contain ocm-repo? just a thought - passing-along repos side-by-side will obviously also work fine.